mat64/matrix.go

```diff
@@ -430,3 +430,23 @@ func use(f []float64, l int) []float64 {
 	}
 	return make([]float64, l)
 }
+
+// useZeroed returns a float64 slice with l elements, using f if it
+// has the necessary capacity, otherwise creating a new slice. The
+// elements of the returned slice are guaranteed to be zero.
+func useZeroed(f []float64, l int) []float64 {
+	if l <= cap(f) {
+		f = f[:l]
+		zero(f)
+		return f
+	}
+	return make([]float64, l)
+}
+
+// zero does a fast zeroing of the given slice's elements.
```

Comment:
This might actually be slower in 1.5  https://groups.google.com/forum/#!topic/golang-codereviews/lx3hsZr4HZE